element_deeplabcut/dlc.py

```diff
@@ -0,0 +1,574 @@
+                      'train_fraction': train_fraction,  'filter_type': filter_type,
+                      'track_method': track_method, 'scorer_legacy': scorer_legacy}
+
+        param_set_hash = dict_to_uuid(param_dict)
+        param_query = cls & {'param_set_hash': param_set_hash}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1('paramset_idx')
+            if skip_duplicates or existing_paramset_idx == paramset_idx:
+                return  # If the existing set has the same paramset_idx: job done
+            else:       # If not: human error adding same paramset with new index
+                raise dj.DataJointError(
+                    'The specified parameter set already exists - paramset_idx: '
+                    + f'{existing_paramset_idx}')
+        else:
+            param_dict.update({'param_set_hash': param_set_hash})
+            cls.insert1(param_dict, skip_duplicates=skip_duplicates)
+
+
+@schema
+class TrainingTask(dj.Manual):
+    definition = """      # Info required to specify 1 model
+    -> VideoRecording     # labeled video for training
+    -> ConfigParamSet
+    training_id: int
+    """
+
+
+@schema
+class ModelTraining(dj.Manual):
+    definition = """
+    -> TrainingTask
+    ---
+    snapshot_index_exact  : int unsigned # latest exact snapshot index (i.e., never -1)
+    config_template       : longblob     # stored full config file
+    """
+
+    @classmethod
+    def train_model(cls, training_id,
+                    max_snapshots_to_keep=5,
+                    displayiters=None,
+                    saveiters=None,
+                    maxiters=None,
+                    allow_growth=False,
+                    gputouse=None,
+                    autotune=False,
+                    keepdeconvweights=True):
+        """ Launches model training, then stores max snapshot number and config
+        :param training_id: Key specifying one entry in dlc.TrainingTask
+        :param project_path: Directory of DLC project
+        :param max_snapshots_to_keep: How many snapshots are kept, default to 5
+        Optional overrides of settings in pose_config.yaml
+            :param displayiters: int, display iterations
+            :param saveiters: int, save X iterations
+            :param maxiters: int, maximum iterations
+        :param allow_growth: bool, default false. True will not pre-allocate gpu memory
+        :param gputouse: int, optional. number of your GPU (see number in nvidia-smi).
+        :param autotune: bool TensorFlow property, faster if 'false'. Default false
+        :param keepdeconvweights: bool, default: true. restore weights from snapshot
+                                  If changing bodyparts, set to true
+        """
+        from deeplabcut import train_network
+        from .readers.dlc_reader import PoseEstimation
+
+        train_key = (TrainingTask & f'training_id={training_id}').fetch1('KEY')

```

Comment:
`training_id` alone is not sufficient to identify one entry in the `TrainingTask`, I'd suggest passing in `training_task_key` as the input argument, instead of just `training_id` and then having to do the restriction/fetch here